knapsack-ampl/main.py

```diff
@@ -96,8 +96,8 @@ def solve(input_data: dict[str, Any], duration: int, provider: str) -> dict[str,
     start_time = time.time()
 
     # Activate license.
-    key = read_key()
-    modules.activate(key)
+    license = read_license_uuid()
+    modules.activate(license)
```

Comment:
I still think we need to handle whether the provider requires a license here. I am getting this when running the base example (this will also be relevant for the auto-tests):

```bash
marius@karma knapsack-ampl$ python main.py -input input.json 
Solving knapsack problem:
  - items: 11
  - capacity: 50
  - max duration: 30 seconds
Traceback (most recent call last):
  File "/home/marius/repos/4_work/community-apps/knapsack-ampl/main.py", line 222, in <module>
    main()
  File "/home/marius/repos/4_work/community-apps/knapsack-ampl/main.py", line 89, in main
    solution = solve(input_data, args.duration, args.provider)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/marius/repos/4_work/community-apps/knapsack-ampl/main.py", line 100, in solve
    modules.activate(license)
  File "/home/marius/.asdf/installs/python/3.12.1/lib/python3.12/site-packages/ampltools/modules/amplpypi.py", line 497, in activate_license
    raise Exception("The license activation failed.")
Exception: The license activation failed.
```

Reply:
Initially, people coming to us are already AMPL users or prospects. I think it is reasonable for commercial-solver apps to require a license to run.

Reply:
I think we need to change the default provider in that case. :sweat_smile: 

More importantly, this will fail the app's README test I am currently bringing back online. This means that we need to skip this app then.

Reply:
That is a good point regarding the test. I would like to have it to make sure we are doing anything blatantly wrong. Ok, I guess we do need that check then.

Reply:
It will also help users switching to free solvers for testing on machines that don't have a license installed. I think it's a nice to have. Just checking the basics should be sufficient as well, as this is still 'just' an example. :blush:

Reply:
Done @merschformann, added a check for the license.

Reply:
Thank you! :hugs: